front/lib/actions/mcp_internal_actions/servers/content_creation/index.ts

```diff
@@ -341,6 +343,66 @@ const createServer = (
     )
   );
 
+  server.tool(
+    RENAME_CONTENT_CREATION_FILE_TOOL_NAME,
+    "Rename a Content Creation file. Use this to change the file name while keeping the content unchanged.",
+    {
+      file_id: z
+        .string()
+        .describe(
+          "The ID of the Content Creation file to rename (e.g., 'fil_abc123')"
+        ),
+      new_file_name: z
+        .string()
+        .describe(
+          "The new name for the file, including extension (e.g., 'UpdatedChart.tsx')"
+        ),
+    },
+    withToolLogging(
+      auth,
+      { toolName: RENAME_CONTENT_CREATION_FILE_TOOL_NAME, agentLoopContext },
+      async ({ file_id, new_file_name }, { sendNotification, _meta }) => {
+        const { agentConfiguration } = agentLoopContext?.runContext ?? {};
+
+        const result = await renameClientExecutableFile(auth, {
+          fileId: file_id,
+          newFileName: new_file_name,
```

Comment:
Shouldn't we replace space with `_` since it's a file name? (generated files usually have no space or with `_`, e.g. `ShibaInuOnePager.tsx`).  For preview title display we use `formatFilenameForDisplay` so we can still show a human friendly title